src/harness/security_testcase.py

```diff
@@ -161,9 +161,9 @@ def doCbsdTestCipher(self, cipher, client_cert=None, client_key=None):
     """
     client_cert = client_cert or self._sas._GetDefaultCbsdSSLCertPath()
     client_key = client_key or self._sas._GetDefaultCbsdSSLKeyPath()
-
+    self._sas.UpdateCbsdRequestUrl(cipher)
```

Comment:
Do you need to call this (or something similar) again at the end of the function? Otherwise I think you may end up with one test case affecting another and the ordering of the tests becomes important.

Reply:
doesn't each test has its own instance of class SasImpl and then its own proper instance url variables?

Reply:
Oops, good point!